### ISSUE TITLE
Add Tendermint 0.34 client support

### ISSUE THREAD

**Issue by webmaster128** [CosmWasm/cosmjs#344]
As of nightly-2020-08-05, Cosmos SDK is still on Tendermint 0.33 (see https://github.com/cosmos/cosmos-sdk/blob/392121e/go.mod#L40). Upgrading to Tendermint 0.34 is planned for the stargate release, but not yet done: https://github.com/cosmos/cosmos-sdk/issues/6365.

We can prepare by creating a Tendermint 0.34 client once there is a release (check https://github.com/tendermint/tendermint/releases, https://hub.docker.com/r/tendermint/tendermint/tags/?page=1&name=v0.34)

**Comment by ethanfrey** [comment]
This may not be necessary for this deliverable. Depends if it is merged into simapp before we deliver. No need to block on it here. We need this for light-client proofs, not the merkle proofs

**Comment by webmaster128** [comment]
Good, thank you. Then we just stop upgrading simapp once https://github.com/cosmos/cosmos-sdk/issues/6365 is done.

**Comment by webmaster128** [comment]
Here is a pending PR with the majority of the Tendermint upgrade, which will probably break interfaces: https://github.com/cosmos/cosmos-sdk/pull/6892 Once this is in, we either need an updated client or pause updating simapp.

**Comment by ethanfrey** [comment]
Good to follow that

**Comment by webmaster128** [comment]
Simapp is based on Tendermint 0.34 starting with `nightly-2020-08-15`. Once we get Tendermint 0.34 RC docker images (https://github.com/tendermint/tendermint/issues/5266) we can start working on a client.

**Comment by ethanfrey** [comment]
Nice one :tada:
